internal/impl/provisioner/endpoint_provisioner_impl.go

```diff
@@ -0,0 +1,471 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package provisioner is defined below
+package provisioner
+
+import (
+	"fmt"
+
+	"solace.dev/go/messaging/internal/ccsmp"
+
+	"solace.dev/go/messaging/internal/impl/constants"
+	"solace.dev/go/messaging/internal/impl/core"
+	"solace.dev/go/messaging/internal/impl/logging"
+	"solace.dev/go/messaging/internal/impl/validation"
+
+	"solace.dev/go/messaging/pkg/solace"
+	"solace.dev/go/messaging/pkg/solace/config"
+)
+
+// endpointProvisionerImpl structure
+type endpointProvisionerImpl struct {
+	properties                  config.EndpointPropertyMap
+	logger                      logging.LogLevelLogger
+	internalEndpointProvisioner core.EndpointProvisioner
+}
+
+// NewEndpointProvisionerImpl function
+func NewEndpointProvisionerImpl(internalEndpointProvisioner core.EndpointProvisioner) solace.EndpointProvisioner {
+	return &endpointProvisionerImpl{
+		// default properties
+		properties:                  constants.DefaultEndpointProperties.GetConfiguration(),
+		logger:                      logging.For(endpointProvisionerImpl{}),
+		internalEndpointProvisioner: internalEndpointProvisioner,
+	}
+}
+
+func mapEndpointPermissionToCcsmpProp(propValue string) string {
+	// the ccsmp permissions map
+	ccsmpPermissionProperties := map[string]string{
+		string(config.EndpointPermissionNone):        string(ccsmp.SolClientEndpointPermissionNone),
+		string(config.EndpointPermissionReadOnly):    string(ccsmp.SolClientEndpointPermissionReadOnly),
+		string(config.EndpointPermissionConsume):     string(ccsmp.SolClientEndpointPermissionConsume),
+		string(config.EndpointPermissionModifyTopic): string(ccsmp.SolClientEndpointPermissionModifyTopic),
+		string(config.EndpointPermissionDelete):      string(ccsmp.SolClientEndpointPermissionDelete),
+	}
+
+	return ccsmpPermissionProperties[propValue]
+}
+
+// validateEndpointProperties function
+func validateEndpointProperties(properties config.EndpointPropertyMap) ([]string, error) {
+	propertiesList := []string{}
+	// We can only provision Queue Endpoints for now; should support Topic Endpoints in the future
+	propertiesList = append(propertiesList, ccsmp.SolClientEndpointPropID, ccsmp.SolClientEndpointPropQueue)
+
+	for property, value := range properties {
+		switch property {
+		case config.EndpointPropertyDurable:
+			isDurable, present, err := validation.BooleanPropertyTruthyValidation(string(config.EndpointPropertyDurable), value)
```

Comment:
The golang api must not be ensuring the durability value as true that should be delegated to ccsmp. We can determine its boolean. But honestly I'm not sure this shuld be a config Property to begin with. It should be set for ccsmp and always true but that is defined by the default properties.